src/include/btmem.h

```diff
@@ -906,6 +910,7 @@ struct __wt_ikey {
  */
 WT_PACKED_STRUCT_BEGIN(__wt_update)
 	uint64_t txnid;			/* transaction */
+	WT_DECL_TIMESTAMP(timestamp)
```

Comment:
The struct is currently padded, and we're expecting timestamps to be 8 bytes in the first real use case (i.e., MongoDB).  We already check that the update structure is the size we expect with a build-time assertion, so I'm inclined to revisit this if we see that assertion fail.  I'm concerned about putting the timestamp *after* the `type` field for the same reason: access to an 8-byte timestamp is more likely to be efficient if it is on an 8 byte boundary.